jobs/docker/templates/bin/ctl

```diff
@@ -84,7 +84,18 @@ case $1 in
         ${DOCKER_USERLAND_PROXY} \
         ${DOCKER_BIP:-} \
         >>${DOCKER_LOG_DIR}/${OUTPUT_LABEL}.stdout.log \
-        2>>${DOCKER_LOG_DIR}/${OUTPUT_LABEL}.stderr.log
+        2>>${DOCKER_LOG_DIR}/${OUTPUT_LABEL}.stderr.log \
+        &
+
+    while [ ! -S ${DOCKER_PID_DIR}/docker.sock ];do
+      sleep .5
+    done
+
+    # Log in to any registries
+<% p('logins', []).each do |login| %>
```

Comment:
Sorry for not thinking of this before as we were discussing different aspects of the PR - I think this setup of `docker login` might go well in a pre-start hook https://bosh.io/docs/post-start.html

As a bonus then we don't need to change the `exec dockerd` as above.

Reply:
BTW, I'm sorry for suggestion subsequent changes. I know it can annoy me when someone doesn't just press the "Merge PR" button ;)

Reply:
No problem - what's right is right so we should just aim for that.

I can do that. I hope to have an update in later today.

Reply:
So things turned out a little funky when moving this to `post-start`. When running in combination with the `containers` job, the start of `containers` fails because it "depends" on the post-start of the `docker` job being successful. Bosh won't move to `post-start` hooks if any of the `start` hooks fail, so we never get to the point that the login happens. Basically, a circular dependency...

I think we may be stuck with this logic in the `start` hook of the `docker` job after all.